File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.java
import java.util.*;
import java.lang.Math;

class Solution {
    public int solution(int[] arr) {
        int answer = arr[0];
        
        for(int i=1; i<arr.length; i++){
            answer = getLcd(answer, arr[i]);
        }
        return answer;
    }
    
    public int getLcd(int a, int b){
        int max = Math.max(a,b);
        int gcd=1;
        
        for(int i=max; i>0; i--){
            if(a % i == 0 && b % i == 0){
                gcd = i;
                break;
            }
        }
        return gcd * (a/gcd) * (b/gcd);
    }
}

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.py
def solution(arr):
    def getGCD(a, b):
        for i in range(b, a*b+1):
            if i % a == 0 and i % b == 0:
                return i
        return 1
    arr.sort()
    answer = getGCD(arr[0], arr[1])
    
    for i in range(2, len(arr)):
        answer = getGCD(answer, arr[i])
    
    return answer  